src/WinRT.Runtime/ObjectReference.cs

```diff
@@ -109,13 +114,25 @@ private protected unsafe IUnknownVftbl VftblIUnknownFromOriginalContext
             }
         }
 
+        [Obsolete]
         protected IObjectReference(IntPtr thisPtr)
         {
             if (thisPtr == IntPtr.Zero)
             {
                 throw new ArgumentNullException(nameof(thisPtr));
             }
             _thisPtr = thisPtr;
+            _contextToken = ComWrappersSupport.IsFreeThreaded(thisPtr) ? IntPtr.Zero : Context.GetContextToken();
+        }
+
+        protected IObjectReference(IntPtr thisPtr, IntPtr contextToken)
```

Comment:
Private protected?

Reply:
Being able to inherit IObjectReference outside WinRT.Runtime allows for custom implementations. I Think that's useful to keep.